### PR TITLE
Add UI plugin to operator

### PR DESCRIPTION
**Pull request by mnecas**
Add forklift ui plugin from https://github.com/kubev2v/forklift-console-plugin This patch adds to the operator:
- RBAC config so the operator have permission to add the ConsolePlugin
- New parameter `feature_ui_plugin` to enable/disable the plugin installation

The UI plugin installation is made of 4 parts:
- ConfigMap with nginx configuration
- Deployment for the forklift UI plugin itself
- Service
- ConsolePlugin

Results:
image

### CODE REVIEW

**Comment by mnecas** [comment]
Currently, the plugin pod fails with:
```

2022/11/07 11:25:14 [error] 7#7: *1 open() "/usr/share/nginx/html/plugin-manifest.json" failed (2: No such file or directory), client: 10.217.0.95, server: , request: "GET /plugin-manifest.json HTTP/1.1", host: "forklift-ui-plugin.konveyor-forklift.svc.cluster.local:9443", referrer: "https://console-openshift-console.apps-crc.testing/dashboards"
10.217.0.95 - - [07/Nov/2022:11:25:14 +0000] "GET /plugin-manifest.json HTTP/1.1" 404 555 "https://console-openshift-console.apps-crc.testing/dashboards" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36"
2022/11/07 11:25:14 [info] 7#7: *1 client 10.217.0.95 closed keepalive connection
```
@sjd78 @yaacov can you please check it out?

File: operator/roles/forkliftcontroller/templates/ui-plugin/configmap-ui-plugin.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    app: {{ app_name }}
+    service: {{ ui_plugin_service_name }}
+  name: nginx-conf
+  namespace: "{{ app_namespace }}"
+data:
+  nginx.conf: "error_log /dev/stdout info;\nevents {}\nhttp {\n\taccess_log         /dev/stdout;\n\tinclude            /etc/nginx/mime.types;\n\tdefault_type       application/octet-stream;\n\tkeepalive_timeout  65;\n\t\tserver {\n\t\t\tlisten              9443 ssl;\n\t\t\tssl_certificate     /var/serving-cert/tls.crt;\n\t\t\tssl_certificate_key /var/serving-cert/tls.key;\n\t\t\troot                /usr/share/nginx/html;\n\t\t}\n\t}\n"
```
Comment (yaacov) [inline operator/roles/forkliftcontroller/templates/ui-plugin/configmap-ui-plugin.yml.j2:11]: can we use a `|` for the multi-line data ? 
```suggestion
  nginx.conf: |
    error_log /dev/stdout info;
    events {}
    ...
```

File: .bazelrc [hunk 2]
```diff
@@ -10,6 +10,7 @@ build --action_env=CONTROLLER_IMAGE=quay.io/konveyor/forklift-controller:latest
 build --action_env=MUST_GATHER_IMAGE=quay.io/konveyor/forklift-must-gather:latest
 build --action_env=MUST_GATHER_API_IMAGE=quay.io/konveyor/forklift-must-gather-api:latest
 build --action_env=UI_IMAGE=quay.io/konveyor/forklift-ui:latest
+build --action_env=UI_PLUGIN_IMAGE=quay.io/kubevirt-ui/forklift-console-plugin:latest
```
Comment (yaacov) [inline .bazelrc:13]: note to self, make sure `latest` is not broken ... :-)

**Review by yaacov — APPROVED** [review]

File: .bazelrc (hunk 2 above)
Comment (mnecas) [reply .bazelrc:13]: well everybody should make that note on all projects + the latest is (or should be) just for devs :D

File: README.md [hunk 3]
```diff
@@ -38,6 +38,7 @@ Another option to override the default values can use `--action_env` as in the e
 | MUST_GATHER_IMAGE     | quay.io/konveyor/forklift-must-gather:latest     | The forklift must gather an image.                          |
 | MUST_GATHER_API_IMAGE | quay.io/konveyor/forklift-must-gather-api:latest | The forklift must gather image api.                         |
 | UI_IMAGE              | quay.io/konveyor/forklift-ui:latest              | The forklift UI image.                                      |
+| UI_PLUGIN_IMAGE       | quay.io/kubevirt-ui/forklift-console-plugin:latest | The forklift UI plugin image.                                      |
```
Comment (ahadas) [inline README.md:41]: let's be more explicit?
```suggestion
| UI_PLUGIN_IMAGE       | quay.io/kubevirt-ui/forklift-console-plugin:latest | The forklift OKD/OpenShift UI plugin image.                                      |
```

File: operator/config/samples/forklift_v1beta1_forkliftcontroller.yaml [hunk 4]
```diff
@@ -6,5 +6,6 @@ metadata:
   namespace: ${NAMESPACE}
 spec:
   feature_ui: 'true'
+  feature_ui_plugin: 'true'
```
[the remark this answers is not in the document]
Comment (mnecas) [reply operator/config/samples/forklift_v1beta1_forkliftcontroller.yaml:9]: Yeah this is just "example". In the forkliftci we disable the UI
https://github.com/kubev2v/forkliftci/blob/6c39a8689f78572f64656f444451bb89f59bb92a/k8s-deploy-forklift.sh#L29-L35

**Comment by yaacov** [comment]
@mnecas @ahadas hi,
https://quay.io/kubev2v/forklift-console-plugin 
is available :tada:

**Review by ahadas — APPROVED** [review]

**Pull request merged**